src/libs/ajax/data-table-providers/EntityServiceDataTableProvider.ts

```diff
@@ -41,4 +55,32 @@ export class EntityServiceDataTableProvider implements DataTableProvider {
   downloadTsv = (signal: AbortSignal, entityType: string): Promise<Blob> => {
     return Ajax(signal).Workspaces.workspace(this.namespace, this.name).getEntitiesTsv(entityType)
   }
+
+  isInvalid: isInvalidFn = (modeMatches: boolean, filePresent: boolean, match: boolean, _: boolean) => {
+    return modeMatches && filePresent && match
+  }
+
+  disabled: disabledFn = (filePresent: boolean, isInvalid: boolean, uploading: boolean, _: boolean) => {
+    return !filePresent || isInvalid || uploading
+  }
+
+  tooltip: tooltipFn = (filePresent: boolean, isInvalid: boolean, _: boolean) => {
+    return !filePresent || isInvalid ? 'Please select valid data to upload' : 'Upload selected data'
+  }
+
+  doUpload: uploadFn = async (uploadParams: UploadTSVParameters) => {
```

Comment:
Is the upload here always going to be a tsv-type file? Wondering if `UploadTSVParameters` could be a misleading interface name here if this function accepts different types of file uploads other than TSV

Reply:
It sure seems like we're All TSV All the Time around here, but maybe it wouldn't hurt to just call it `UploadParameters`.